pkg/operator/ceph/spec/spec.go

```diff
@@ -62,6 +65,57 @@ func RookVolumeMounts() []v1.VolumeMount {
 	)
 }
 
+// DaemonVolumes returns the pod volumes used by all Ceph daemons.
+func DaemonVolumes(dataPaths *config.DataPathMap, keyringResourceName string) []v1.Volume {
+	var dataDirSource v1.VolumeSource
+	if dataPaths.PersistData {
+		dataDirSource = v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: dataPaths.HostDataDir}}
+	} else {
+		dataDirSource = v1.VolumeSource{EmptyDir: &v1.EmptyDirVolumeSource{}}
+	}
+	return []v1.Volume{
+		{Name: "ceph-daemon-data", VolumeSource: dataDirSource},
+		config.ConfigVolume(),
+		config.KeyringVolume(keyringResourceName),
+	}
+}
+
+// DaemonVolumeMounts returns volume mounts which correspond to the DaemonVolumes. These
+// volume mounts are shared by most all Ceph daemon containers, both init and standard.
+func DaemonVolumeMounts(dataPaths *config.DataPathMap, keyringResourceName string) []v1.VolumeMount {
+	return []v1.VolumeMount{
+		{Name: "ceph-daemon-data", MountPath: dataPaths.ContainerDataDir},
+		config.ConfigVolumeMount(),
+		config.KeyringVolumeMount(keyringResourceName),
+	}
+}
+
+// DaemonFlags returns the command line flags used by all Ceph daemons.
+func DaemonFlags(
+	clusterInfo *cephconfig.ClusterInfo,
+	daemonType config.DaemonType, daemonID string,
+) []string {
+	c := config.NewConfig()
+	c.Section("global").
+		// fsid unnecessary but is a safety to make sure daemons can only connect to their cluster
+		Set("fsid", clusterInfo.FSID).
+		// all daemon names are of the form "<type>.<id>"
+		Set("name", fmt.Sprintf("%s.%s", string(daemonType), daemonID)).
+		Set("keyring", config.ContainerKeyringFilePath())
+	return append(
```

Comment:
All ceph daemons share 5 flags: `fsid`, `name=<dmn>.<id>`, `keyring`, `mon_host`, and `mon_initial_members`. All other configs are set in the config file currently but in the future will be set in the mon's kv store for mimic+.